### ISSUE TITLE
Bug: route group filter bug

### ISSUE THREAD

**Issue by sajibAdhi** [codeigniter4/CodeIgniter4#7963]
### PHP Version

8.2

### CodeIgniter4 Version

4.4.1

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows, Linux

### Which server did you use?

apache

### Database

MySQL 5.6

### What happened?

I was using filters in `Grouping Routes` which was working properly in `nested routes` and `nested grouping routes`. But when I used a namespace in `nested grouping routes` the filter stopped working for `nested grouping routes`.

### Steps to Reproduce

```php
$routes->group('', ['filter' => AuthenticationFilter::class], function ($routes) {
    $routes->get('dashboard', function (){
        echo "AuthenticationFilter is working";
    });
    
    $routes->group('profile', ['namespace' => 'App\Controllers\Profile'], function($routes){
        $routes->get('/', function (){
            echo "AuthenticationFilter is not working";
        });
    });
});
```

### Expected Output

I am expecting the auth filter to be called before the profile request.

**Comment by kenjis** [comment]
> Note
> Options passed to the outer group() (for example namespace and filter) are not merged with the inner group() options.
https://codeigniter4.github.io/CodeIgniter4/incoming/routing.html#nesting-groups

**Comment by sajibAdhi** [comment]
> > Note
> > Options passed to the outer group() (for example namespace and filter) are not merged with the inner group() options.
> > https://codeigniter4.github.io/CodeIgniter4/incoming/routing.html#nesting-groups

Oh!! 😭 😭 It's kind of pain.

**Comment by kenjis** [comment]
But the current behavior is confusing.

```php
$routes->group('', ['filter' => 'csrf'], function ($routes) {
    $routes->get('dashboard', function (){
        echo "CSRF is working";
    });

    $routes->group('profile', function($routes){
        $routes->get('/', function (){
            echo "CSRF is working";
        });
    });
});
```

```php
$routes->group('', ['filter' => 'csrf'], function ($routes) {
    $routes->get('dashboard', function (){
        echo "CSRF is working";
    });

    $routes->group('profile', [], function($routes){
        $routes->get('/', function (){
            echo "CSRF is not working";
        });
    });
});
```

**Comment by kenjis** [comment]
@codeigniter4/core-team 
Why the options array passed to the outer group() are not merged with the inner group() options array?
https://github.com/codeigniter4/CodeIgniter4/blob/c4bd34e48ba20be97c92357f3e0d27552aa8cb96/system/Router/RouteCollection.php#L774

**Comment by michalsn** [comment]
Dunno, sounds like a bug... even though it's documented.

**Comment by kenjis** [comment]
I would like to fix as a bug, but the impact may be big.
So I sent PR to `4.5` branch. See #7981

**Comment by MGatner** [comment]
Followed up on the PR.

**Comment by neznaika0** [comment]
A small question on the topic. Is it normal that the "filter" option is applied before and after the request?

```php
$routes->group('', ['filter' => 'auth'], static function ($routes) {
    $routes->get('/expenses', 'Expense::list', ['as' => 'expenses.list']);
});
```
```console
+--------+--------------------+--------------------+-------------------------------------+----------------+---------------+
| Method | Route              | Name               | Handler                             | Before Filters | After Filters |
+--------+--------------------+--------------------+-------------------------------------+----------------+---------------+
| GET    | expenses           | »                  | \App\Controllers\Expense::list      | auth csrf      | auth          |
```

**Comment by kenjis** [comment]
Yes, it is expected, but probably the auth filter does nothing in `after()`.

**Comment by neznaika0** [comment]
Good. What will happen if the filter changes something or does more complicated work?

**Comment by kenjis** [comment]
I sent another PR #8033

**Comment by kenjis** [comment]
Hi all, I created two PRs.
Please approve whichever you think is better.
- #7981
- #8033

**Comment by kenjis** [comment]
Closed by #8033
